source/server/listener_socket_option_impl.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <netinet/tcp.h>
+
+#include "envoy/api/v2/listener/listener.pb.h"
+
+#include "common/network/listen_socket_impl.h"
+#include "common/network/socket_option_impl.h"
+
+#include "server/configuration_impl.h"
+
+#ifdef TCP_FASTOPEN
+#define ENVOY_SOCKET_TCP_FASTOPEN Network::SocketOptionName(TCP_FASTOPEN)
+#else
+#define ENVOY_SOCKET_TCP_FASTOPEN Network::SocketOptionName()
```

Comment:
Have you tried manually forcing this option and seeing what happens to the tests? In my experience, this is a useful step.

Reply:
I mean as an experiment in your local client, change that `#ifdef` line to `#if 0` and see if `bazel test //test/...` passes.